### PR TITLE
Add a `sindri project delete` command

### PR DESCRIPTION
**Pull request by sangaline**
This adds a new `sindri project delete` subcommand which can be used to delete either a specific project or all projects recursively. The main motivation is that we record test fixtures against the production API and they've been accumulating a lot of cruft over time, so now we can call `sindri project delete --all` on CI to get a clean slate before recording new test fixtures.

There's starting to be some code duplication in the CLI commands around things like finding and parsing a project's `sindri.json` file. We should aim to refactor and clean some of this up, but I think it's manageable for now.

### CODE REVIEW

File: src/cli/project.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        sindri.logger.warn("Aborting. No projects were deleted.");
+        return;
+      }
+    }
+
+    // Perform the deletion.
+    let deletedCount = 0;
+    if (!all) {
+      name = name || findProjectName(directory);
+      await deleteProject(name);
```
Comment (katiemckeon) [inline src/cli/project.ts:149]: Off by one in this case (so the log would say "...deleted 0 projects")
```suggestion
      await deleteProject(name);
      deletedCount++;
```

File: src/cli/project.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+const projectDeleteCommand = new Command()
+  .name("delete")
+  .description("Delete a project and all associated builds and proofs.")
+  .option(
+    "-a, --all",
+    "Delete *all* projects. Overrides the `--name` and directory options.",
+  )
+  .option("-f, --force", "Skip the confirmation prompt.")
```
Comment (katiemckeon) [inline src/cli/project.ts:119]: Nice addition.

File: src/cli/project.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    return process.exit(1);
+  }
+  if (!("name" in sindriJson) || typeof sindriJson.name !== "string") {
+    sindri.logger.error('No "name" field found in "sindri.json". Aborting.');
+    return process.exit(1);
+  }
+  return sindriJson.name;
+}
+
+async function getSomeProjectNames(teamName: string): Promise<string[]> {
```
Comment (katiemckeon) [inline src/cli/project.ts:76]: I don't feel passionate about this suggestion, but my first readthrough of the code misinterpreted why this function is here. I thought you were laying the scaffolding out for future `project` subcommands.  
```suggestion
async function getSomeProjectNames(teamName: string): Promise<string[]> {
  // Used as an efficient iterator for `projectDeleteCommand`.
```

**Review by katiemckeon — COMMENTED** [review]
Nice new functionality!  I trimmed a lot of bloat while testing this.

## Tests
To get a feel for logging and edge cases I tested:
* happy path (one named project with many circuits was successfully axed)
* delete a project by name that doesn't exist
* delete from a directory with a sindri.json but doesn't actually correspond to Sindri project
* delete from a directory with no manifest

I did not test the delete all command, cause I do not have an appropriate sacrifice. But I carefully reviewed and approve the logic there

## Comment

> We should aim to refactor and clean some of this up

You mention code duplication.  To me that is a lesser concern versus the "project build" vs "circuit" confusion that is mounting in these CLI docstrings. When we produce the docs for this we will have a situation like:
>   `clone <circuit> [directory]   Clone a circuit into a local directory.`
>  `deploy [options] [directory]  Deploy the current Sindri project.`
>  `project                       Commands related to managing projects.`

As an observer of just the docs, I'd wonder why `deploy` was not nested under `project`. And when I see that `project delete` will remove all "project builds", I would not be certain that "circuits" (which I have been cloning left and right from other teams) are part of that or a separate kind of thing.

Not an immediate request for a fix - but I think external clarity should be kept as a priority if we undergo a refactor

**Review by katiemckeon — APPROVED** [review]
LGTM!
